crates/core/src/state_dump.rs

```diff
@@ -41,8 +43,9 @@ impl From<Arc<Context>> for StateDump {
         let source_chain: Vec<ChainHeader> = source_chain.into_iter().rev().collect();
 
         let queued_calls: Vec<ZomeFnCall> = nucleus.queued_zome_calls.into_iter().collect();
-
         let running_calls: Vec<ZomeFnCall> = nucleus.running_zome_calls.into_iter().collect();
+        let call_results: Vec<(ZomeFnCall, Result<_, _>)> =
+            nucleus.zome_call_results.into_iter().collect();
```

Comment:
I just noticed that we're now storing all zome call results in the state. I'm concerned about this growing huge for zome calls that return get_links. Imagine a DNA with hundreds of users with a UI that polls a get_all_messages zome function periodically...

Reply:
Yep, but this is not new. So far this wasn't a problem (as the zome call results do net get persisted in state snapshots) but I agree we should add a way of clearing that Vec, or transform it into a ring buffer - but in a separate PR.